tfe/resource_tfe_notification_configuration.go

```diff
@@ -0,0 +1,205 @@
+						[]string{
+							string(tfe.NotificationTriggerCreated),
+							string(tfe.NotificationTriggerPlanning),
+							string(tfe.NotificationTriggerNeedsAttention),
+							string(tfe.NotificationTriggerApplying),
+							string(tfe.NotificationTriggerCompleted),
+							string(tfe.NotificationTriggerErrored),
+						},
+						false,
+					),
+				},
+			},
+
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"workspace_external_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFENotificationConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get workspace
+	workspaceID := d.Get("workspace_external_id").(string)
+
+	// Get attributes
+	destinationType := tfe.NotificationDestinationType(d.Get("destination_type").(string))
+	enabled := d.Get("enabled").(bool)
+	name := d.Get("name").(string)
+	token := d.Get("token").(string)
+	url := d.Get("url").(string)
+
+	// Throw error if token is set with destinationType of slack
+	if token != "" && destinationType == tfe.NotificationDestinationTypeSlack {
+		return fmt.Errorf("token cannot be set with destination_type of %s", destinationType)
+	}
+
+	// Create a new options struct
+	options := tfe.NotificationConfigurationCreateOptions{
+		DestinationType: tfe.NotificationDestination(destinationType),
+		Enabled:         tfe.Bool(enabled),
+		Name:            tfe.String(name),
+		Token:           tfe.String(token),
+		URL:             tfe.String(url),
+	}
+
+	// Add triggers set to the options struct
+	for _, trigger := range d.Get("triggers").(*schema.Set).List() {
+		options.Triggers = append(options.Triggers, trigger.(string))
+	}
+
+	log.Printf("[DEBUG] Create notification configuration: %s", name)
+	notificationConfiguration, err := tfeClient.NotificationConfigurations.Create(ctx, workspaceID, options)
+	if err != nil {
+		return fmt.Errorf("Error creating notification configuration %s: %v", name, err)
+	}
+
+	d.SetId(notificationConfiguration.ID)
+
+	return resourceTFENotificationConfigurationRead(d, meta)
+}
+
+func resourceTFENotificationConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	log.Printf("[DEBUG] Read notification configuration: %s", d.Id())
+	notificationConfiguration, err := tfeClient.NotificationConfigurations.Read(ctx, d.Id())
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			log.Printf("[DEBUG] Notification configuration %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading notification configuration %s: %v", d.Id(), err)
+	}
+
+	// Update config
+	d.Set("destination_type", notificationConfiguration.DestinationType)
+	d.Set("enabled", notificationConfiguration.Enabled)
+	d.Set("name", notificationConfiguration.Name)
+	// Don't set token here, as it is write only
+	// and setting it here would make it blank
```

Comment:
Thank you for adding this comment! I noticed that resources like `OrganizationToken` and `TeamToken` also don't set the actual token value, but with no comment as to why.

Reply:
I pieced this together after a lot of trial and error. It would be nice to get confirmation from someone who has done a lot of work with providers though.